yugabyte/run-jepsen.py

```diff
@@ -55,68 +65,172 @@
+    run_cmd(SORT_RESULTS_SH)
+
+    start_time = time.time()
+    nemeses = NEMESES
+    if args.enable_clock_skew:
+        nemeses += ['clock-skew']
+
+    num_tests_run = 0
+    total_test_time_sec = 0
+
+    is_done = False
+    if os.path.isdir(LOGS_DIR):
+        logging.info("Directory %s already exists", LOGS_DIR)
     else:
-        timed_out = False
-
-    child_processes.remove(p)
-    if exit_on_error and p.returncode != 0:
-        sys.exit(p.returncode)
-    return CmdResult(returncode = p.returncode, timed_out = timed_out)
-
-atexit.register(cleanup)
-
-run_cmd(SORT_RESULTS_SH)
-
-while True:
-    for nemesis in NEMESES:
-        for test in TESTS:
-            result = run_cmd(
-                "lein run test --os debian --url {URL} --workload {test} --nemesis {nemesis} --concurrency 5n "
-                "--time-limit {run_time}".format(**locals()),
-                timeout=TIMEOUT, run_time=SINGLE_TEST_RUN_TIME, exit_on_error=False
-            )
-            if result.timed_out:
-                for root, dirs, files in os.walk(STORE_DIR):
-                    for file in files:
-                        if file == "jepsen.log":
-                            msg = "Test run timed out!"
-                            print "\n" + msg
-                            with open(os.path.join(root, file), "a") as f:
-                                f.write(msg)
-
-            run_cmd(SORT_RESULTS_SH)
-
+        logging.info("Creating directory %s", LOGS_DIR)
+        os.mkdir(LOGS_DIR)
+
+    while not is_done:
+        for nemesis in nemeses:
+            if is_done:
+                break
+            for test in TESTS:
+                elapsed_time_sec = time.time() - start_time
+                if args.max_time_sec is not None and elapsed_time_sec > args.max_time_sec:
+                    logging.info(
+                        "Elapsed time is %.1f seconds, it has exceeded the max allowed time %.1f, "
+                        "stopping", elapsed_time_sec, args.max_time_sec)
+                    is_done = True
+                    break
+
+                test_start_time_sec = time.time()
+                result = run_cmd(
+                    "lein run test "
+                    "--os debian "
+                    "--url {url} "
+                    "--workload {test} "
+                    "--nemesis {nemesis} "
+                    "--concurrency 5n "
+                    "--time-limit {run_time}".format(
+                        url=args.tarball_url,
+                        test=test,
+                        nemesis=nemesis,
+                        run_time=SINGLE_TEST_RUN_TIME
+                    ),
+                    timeout=TEST_TIMEOUT,
+                    exit_on_error=False,
+                    log_name_prefix="{}_nemesis_{}".format(test, nemesis)
+                )
+
+                if result.timed_out:
+                    logging.info("Test timed out. Updating all jepsen.log files in %s", STORE_DIR)
```

Comment:
Actually, we need to update just the current one which is in `store/current/jepsen.log`, so this could be simplified. `store/current` is a symlink to a directory with the current test run.